rl_equation_solver/utilities/reward.py

```diff
@@ -0,0 +1,91 @@
+"""Collection of reward functions"""
+from abc import abstractmethod
+import numpy as np
+
+
+class RewardMixin:
+    """Reward function collection"""
+
+    @abstractmethod
+    def expression_complexity(self, state):
+        """Get the graph / expression complexity for a given state"""
```

Comment:
where graph complexity := number of nodes + number of edges. Example:

The graph for ax + b = 0 is:

        =
       /   \
      add  0
    /   \
   mult  b
 /      \
a     x

which has 7 nodes + 6 edges = complexity of 13

Reply:
Shoot, the graph above isn't rendering